index.js

```diff
@@ -144,6 +145,57 @@ function reset () {
   moduleAliasNames = []
 }
 
+//
+// Functions to load configurations from a file
+//
+
+// Load configuration from any file
+function loadConfigFile (configPath) {
+  try {
+    var config = require(configPath)
+
+    // ES6 export default
+    if (config && config.default) {
+      config = config.default
+    } else if (config && config['module-alias']) {
+      config = config['module-alias']
+    }
+
+    return config
+  } catch (e) {
+    // Do nothing
+  }
+}
+
+function loadConfig (base, options) {
```

Comment:
Options is not used here

```suggestion
function loadConfig (base) {
```

Reply:
Hey @Kehrlann , Thank you for the kind words.

But I've found two issues while experimenting with a (soft) symbolic path to a dir. It has nothing to do with `fs.lstatSync`.

**What I did**
1. I moved the `/test/src/default_module_alias` dir to somewhere else in my PC and created a soft symlink in `/test/src/default_module_alias`. 

2. It works, and I can access the files as same as before without any issue. 

3. I ran the test and 3 tests were failing. And I found out that if I change the current working directory to `/test/src/default_module_alias` and does not pass a `base` path. It uses `process.cwd` which is as expected.

4. `process.cwd` returns the absolute path, `/home/username/Desktop/secret/default_module_alias` instead of `<redacted>/test/src/default_module_alias`, which is also an expected behaviour. 

5. The issue arises when the configuration file has entry to go up (`../`) in the directory.
```
module.exports = {
  aliases: {
    '@src': '../',
    '@foo': '../foo/index.js',
    '@bar': '../bar',
    'some/foo': '../foo'
  },
  moduleDirectories: ['../node_modules_custom']
}
``` 
This produces `/home/username/Desktop/secret/foo/index.js` instead of `/test/src/foo/index.js`, so file not found and 3 tests fails.

In the rest function, if I change
```
Object.getOwnPropertyNames(require.cache).forEach(function (name) {
      if (name.indexOf(path) !== -1) {
        delete require.cache[name]
      }
})
```
to
```
Object.getOwnPropertyNames(require.cache).forEach(function (name) {
        delete require.cache[name]
})
```
The number of failing tests drops to only 1 since the modules were cached.

Now, I'm wondering what are some possible cases where the user has the configuration file in a symlink dir?

Please let me know if it does not make sense, I'll try to change some words, or maybe push the work so you can test it yourself.

Reply:
Thanks for the detailed explanation !

Actually now that you talk about it, I'm not sure symlinking your config file itself sounds like a corner case. Not sure we need to worry about it too much ; maybe just put a note in the docs about it.